backend/dataall/modules/worksheets/services/worksheet_query_result_service.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import io
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from openpyxl import Workbook
+
+from dataall.base.aws.s3_client import S3_client
+from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.modules.worksheets.db.worksheet_models import WorksheetQueryResult
+from dataall.modules.worksheets.db.worksheet_repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_service import WorksheetService
+
+if TYPE_CHECKING:
+    try:
+        from sqlalchemy.orm import Session
+        from mypy_boto3_s3.client import S3Client
+    except ImportError:
+        print('skipping type checks as stubs are not installed')
+        S3Client = None
+        Session = None
+
+
+class WorksheetQueryResultService:
+    SupportedFormats = {'csv', 'xlsx'}
```

Comment:
Let's use an Enum for the supported formats to avoid string issues. To avoid circular dependencies with `api`, we should move `dataall/modules/worksheets/api/enums.py` to `backend/dataall/modules/worksheets/services/worksheet_enums.py` and add
```
class WorksheetResultsFormat(GraphQLEnumMapper:
    CSV = 'csv'
    XLSX= 'xlsx'
```

Reply:
check the comment in [backend/dataall/modules/worksheets/services/worksheet_query_result_service.py](https://github.com/data-dot-all/dataall/pull/1547/files/01d681db39722f3ad04493271cf2b02533789e92#diff-cdc94a4ea25d93c1b5b7ae5bf723c13b0b1fcb2e9dd392764ec0645b655c039c)